compiler/src/codegen/transl_anf.re

```diff
@@ -539,24 +539,32 @@ let compile_imm = (env, i: imm_expression) =>
   | ImmTrap => MImmTrap
   };
 
+type known_function =
+  | Internal(Ident.t)
+  | Imported(Ident.t, string);
+
 let known_functions = Ident_tbl.create(50);
-let register_function = id => Ident_tbl.add(known_functions, id, ());
+let register_function = func =>
+  switch (func) {
+  | Internal(id)
+  | Imported(id, _) => Ident_tbl.add(known_functions, id, func)
+  };
 let clear_known_functions = () => Ident_tbl.clear(known_functions);
 let known_function = f =>
   switch (f.imm_desc) {
   | ImmId(id) =>
-    if (Ident_tbl.mem(known_functions, id)) {
-      Some(Ident.unique_name(id));
-    } else {
-      None;
+    switch (Ident_tbl.find_opt(known_functions, id)) {
+    | Some(Internal(id)) => Some(Ident.unique_name(id))
+    | Some(Imported(id, name)) => Some(name)
```

Comment:
double-checking: `name` is unique here (defined at the import site), right?

Reply:
Yup, unique by module/function pair.